Better recovery for the Fulltext Index Add-On

chrisvest opened the pull request:
Based on #10060

chrisvest committed:
Only drop and rebuild fulltext indices if needed.

ragadeeshu committed:
Test repopulate fulltext index on analyzer change

ragadeeshu committed:
Update fulltext addon test after rebase

MishaDemianenko commented:
@chrisvest needs rebase

chrisvest commented:
@MishaDemianenko rebased.

MishaDemianenko commented on enterprise/fulltext-addon/src/main/java/org/neo4j/kernel/api/impl/fulltext/integrations/bloom/BloomKernelExtension.java:
```diff
@@ -72,14 +70,14 @@ public void init() throws IOException, KernelException
-            List<String> properties = getProperties();
-            Analyzer analyzer = getAnalyzer();
+            List<String> properties = getProperties( );
+        String analyzer = config.get( LoadableBloomFulltextConfig.bloom_analyzer);
```
a bit of formatting here

chrisvest committed:
Rebuild the index if the last committed transaction id has changed since the index was last shut down.

This can happen if the index has been disabled for a while and new data has been committed during this time.

chrisvest committed:
Remove unnecessary test code from the fulltext updater tests
